### PR TITLE
Moved over tamu customizations

### PR DESCRIPTION
**Pull request by jeremythuff**
Migrates TAMU customizations into the Nolana release.

### CODE REVIEW

File: src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java [hunk 1]
```diff
@@ -275,6 +282,69 @@ private void updateSubfieldsMapWithItemLoanTypeSubfield(Map<String, Object> subF
...
+   *
+   * @param jsonData                    - json of single item or holding
+   * @param marcRecordFields            - fields list to be updated with new one
+   * @param suppressedRecordsProcessing - include suppressed flag in 999 field?
+   */
+  private void updateFieldsWithHoldingsRecordField(JsonObject jsonData,
+                                                   List<Object> marcRecordFields,
+                                                   boolean suppressedRecordsProcessing) {
+    Map<String, Object> holdingsRecordSubFields = constructHoldingsRecordSubFieldsMap(jsonData);
+    int subFieldValue = BooleanUtils.isFalse(jsonData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD)) ? 0 : 1;
```
Comment (ghost) [inline src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: Should we use `calculateDiscoverySuppressedSubfieldValue` method here?

```
  /**
   * Calculates discovery suppressed subfield value.
   *
   * @param itemData - json of single item
   * @return subfield value
   */
  private int calculateDiscoverySuppressedSubfieldValue(JsonObject itemData) {
    if (isNull(itemData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD))) {
      return 0;
    }
    return BooleanUtils.isFalse(itemData.getBoolean(INVENTORY_SUPPRESS_DISCOVERY_FIELD)) ? 0 : 1;
  }
  ```

File: src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java [hunk 2]
```diff
@@ -398,6 +468,8 @@ private enum EffectiveLocationSubFields {
       this.jsonPropertyPath = jsonPropertyPath;
     }
 
+
```
Comment (ghost) [inline src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: Some extraneous newlines.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java [hunk 3]
```diff
@@ -76,7 +76,7 @@ class RecordMetadataManagerTest {
   private RecordMetadataManager metadataManager = RecordMetadataManager.getInstance();
   private StorageHelper storageHelper = new SourceRecordStorageHelper();
 
-  @Test
+  //@Test
```
Comment (ghost) [inline src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: What fails for these? Do they pass without our customizations? Can we not adjust for them to pass? If all else, @Ignore would be better than commenting them out.

Comment (jeremythuff) [reply src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: I am not sure if they fail, I was bringing this change forward. I saw it initially [here](https://github.com/TAMULib/mod-oai-pmh/pull/5/files#diff-88314c7df6e154e36d1ed77b9358f55d8cab340bf5102ee48f3aa9e42eee1466R75)

After a little testing, it appears there are assertion errors when they are run. The error is occurring with @Ignore as well.

Comment (jeremythuff) [reply src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: @Ignore resulted in the same error, @Disabled seemed to work

File: src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java (hunk 1 above)
Comment (jeremythuff) [reply src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: It does not appear that they are doing this [upstream](https://github.com/folio-org/mod-oai-pmh/blob/v3.10.0/src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java#L219), so I vote "no".

Comment (ghost) [reply src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: It is an additional null check and that is just a place missed to reduce redundancy. I suggest we do not do the same and use the method created to perform the same logic in one place with the additional null check.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java (hunk 3 above)
Comment (ghost) [reply src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: Yes, junit5 uses @disabled. That was an older snapshot than the 3.8.0 image used last.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java [hunk 4]
```diff
@@ -76,7 +77,7 @@ class RecordMetadataManagerTest {
...
```
Comment (ghost) [inline src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: Missed one. Can we see the failures and determine is simple change to assertion could fix? Or is this issue timeboxed?

File: src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java [hunk 5]
```diff
@@ -275,6 +282,69 @@ private void updateSubfieldsMapWithItemLoanTypeSubfield(Map<String, Object> subF
...
+    if(Objects.nonNull(holdingsData)) {
+      String subFieldCode = subFieldGroupProperty.getSubFieldCode();
+      String subFieldValue = holdingsData.getString(subFieldGroupProperty.getJsonPropertyPath());
+      if (isNotEmpty(subFieldValue)) {
+        effectiveLocationSubFields.put(subFieldCode, subFieldValue);
+      }
+    }
+  }
+
+
```
Comment (ghost) [inline src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: Extraneous newline.

File: src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java [hunk 6]
```diff
@@ -172,16 +176,19 @@ private void updateFieldsWithItemEffectiveLocationField(JsonObject itemData,
                                                           List<Object> marcRecordFields,
                                                           boolean suppressedRecordsProcessing) {
     Map<String, Object> effectiveLocationSubFields = constructEffectiveLocationSubFieldsMap(itemData);
+    int subFieldValue = calculateDiscoverySuppressedSubfieldValue(itemData);
     if (suppressedRecordsProcessing) {
-      effectiveLocationSubFields.put(DISCOVERY_SUPPRESSED_SUBFIELD_CODE, calculateDiscoverySuppressedSubfieldValue(itemData));
+      effectiveLocationSubFields.put(DISCOVERY_SUPPRESSED_SUBFIELD_CODE, subFieldValue);
+    }
+    if(subFieldValue == 0) {
```
Comment (ghost) [inline src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java]: Could use space between `if` and `(`.

File: src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java (hunk 4 above)
Comment (jeremythuff) [reply src/test/java/org/folio/oaipmh/helpers/records/RecordMetadataManagerTest.java]: It is to some extent, we are waiting on this deployment to get to the some of the QA

**Review by ghost — APPROVED** [review]

**Review by kaladay — APPROVED** [review]

**Comment by tamu-sad-iii** [comment]
@jeremythuff is this PR still required or are we now running non custom mod-oai-pmh?

**Review by rmathew1011 — APPROVED** [review]

**Comment by kaladay** [comment]
A newer version is in use, this PR is no longer needed.

**Pull request closed without merging**